pkg/public/api.go

```diff
@@ -11,7 +11,7 @@ import (
 // plane. If the checks fail, then CLI will print an error and exit. If the
 // hcOptions.retryDeadline param is specified, then the CLI will print a
 // message to stderr and retry.
-func CheckPublicAPIClientOrRetryOrExit(hcOptions healthcheck.Options) {
+func CheckPublicAPIClientOrRetryOrExit(hcOptions healthcheck.Options) *healthcheck.HealthChecker {
```

Comment:
Could we keep these separate? Maybe we introduce `func NewPublicAPIClient` and then it gets initialized by a call to `func (hc *healthcheck.HealthChecker) CheckPublicAPIClient`. Afterwards we should still be able to check `hc.cniEnabled`.

It adds a little code, but here when we only care about checking certain things we don't really need to return anything.

Reply:
Agreed adding a return there felt a little weird. I've pushed e3a54506c which separates these things, just for the the viz install as an example for now. Those new functions would live directly inside the `healthcheck` package, without having to go through the intermediary `public` package (`pkg/public/api.go`) which we can remove after applying this to all the extensions.

Reply:
Yea I like this. So this would then allow us to remove `func CheckPublicAPIClientOrRetryOrExit` and `func exitOnError`.

Once this is used in the other related changes in this PR, it should look good to me.

Reply:
Have to say, it looks really nice this way.